File: 백준/Silver/1920. 수 찾기/수 찾기.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

class Main {


    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int[] arr = new int[N];


        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr);
        
        st = new StringTokenizer(br.readLine());

        int M = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < M; i++) {
            boolean find = false;
            int target = Integer.parseInt(st.nextToken());
            int start = 0;
            int end = arr.length - 1;

            while (start <= end) {
                int mid = (start + end) / 2;
                int midVal = arr[mid];
                if(midVal > target) {
                    end = mid - 1;
                } else if(midVal < target) {
                    start = mid + 1;
                } else {
                    find = true;
                    break;
                }
            }

            if(find) System.out.println(1);
            else System.out.println(0);
        }
    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {

    var br = BufferedReader(InputStreamReader(System.`in`))
    var st = StringTokenizer(br.readLine())

    val N = st.nextToken().toInt()

    st = StringTokenizer(br.readLine())

    var firstArr = arrayOfNulls<Int>(N)

    for (i in 0 until N ) {
        firstArr[i] = st.nextToken().toInt()
    }

    firstArr.sort() // 1,2,3,4,5

    st = StringTokenizer(br.readLine())

    var M = st.nextToken().toInt()

    var secondArr = arrayOfNulls<Int>(M)

    st = StringTokenizer(br.readLine())
    for (i in 0 until M) {
        secondArr[i] = st.nextToken().toInt()
    }

    for (i in 0 until M) {
        var start: Int = 0
        var end: Int = firstArr.size - 1
        var find: Boolean = false
        while (start <= end) {
            var mid: Int = (end + start) / 2
            var mid_value: Int? = firstArr[mid]

            if(mid_value!! > secondArr[i]!!) {
                end = mid - 1
            }else if (mid_value < secondArr[i]!!) {
                start = mid + 1
            } else {
                find = true
                break
            }
        }

        if (find) {
            println(1)
        } else {
            println(0)
        }
    }






}